### PR TITLE
Fail on invalid and unknown ROS specific arguments

### PR DESCRIPTION
**Pull request by hidmic**
Precisely what the title says. This pull request makes `rclcpp` throw if either an invalid or an unknown ROS argument is found.

### CODE REVIEW

File: rclcpp/include/rclcpp/utilities.hpp [hunk 1]
```diff
@@ -282,6 +284,27 @@ sub_will_underflow(const T x, const T y)
   return (y > 0) && (x < (std::numeric_limits<T>::min() + y));
 }
 
+/// Join values in a container turned into strings by a given delimiter
```
Comment (ivanpauno) [inline rclcpp/include/rclcpp/utilities.hpp]: I would rather add this in `rcpputils`. We should also move some stuff here to that package.

File: rclcpp/include/rclcpp/utilities.hpp [hunk 2]
```diff
@@ -15,6 +15,7 @@
 #ifndef RCLCPP__UTILITIES_HPP_
 #define RCLCPP__UTILITIES_HPP_
 
+
```
Comment (ivanpauno) [inline rclcpp/include/rclcpp/utilities.hpp]: nit: unneeded new line

File: rclcpp/include/rclcpp/utilities.hpp [hunk 3]
```diff
@@ -26,7 +27,6 @@
 #include "rclcpp/visibility_control.hpp"
 
 #ifdef ANDROID
-#include <sstream>
```
Comment (ivanpauno) [inline rclcpp/include/rclcpp/utilities.hpp]: Why was this deleted?

File: rclcpp/include/rclcpp/utilities.hpp (hunk 2 above)
Comment (hidmic) [reply rclcpp/include/rclcpp/utilities.hpp]: Leftover of when `join()` was here. Reverted.

File: rclcpp/include/rclcpp/utilities.hpp (hunk 3 above)
Comment (hidmic) [reply rclcpp/include/rclcpp/utilities.hpp]: Leftover of when `join()` was here. Reverted.

File: rclcpp/include/rclcpp/utilities.hpp [hunk 4]
```diff
@@ -27,7 +27,6 @@
 
 #ifdef ANDROID
 #include <sstream>
-
```
Comment (ivanpauno) [inline rclcpp/include/rclcpp/utilities.hpp]: nit: There's now an extra line deletion in `utilities.hpp` :P

Comment (hidmic) [reply rclcpp/include/rclcpp/utilities.hpp]: Argh! Fixed.
